code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/azure/AzureTokenCreator.java

```diff
@@ -51,4 +51,24 @@ public void reset() {
     public String getName() {
         return AzureConstants.TOKEN_CREATOR;
     }
+
+    @Override
+    public Token getCurrentToken() {
```

Comment:
Each TokenCreator is having exact same implementation of this method. Instead add this logic to AbstractTokenUtil::readCurrentToken() and invoke that from each tokenCreator:: getCurrentToken()